app/gui/view/debug_scene/cloud-dashboard/src/lib.rs

```diff
@@ -0,0 +1,680 @@
+#![feature(fn_traits)]
+#![feature(marker_trait_attr)]
+#![feature(specialization)]
+#![feature(trait_alias)]
+#![feature(type_alias_impl_trait)]
+#![feature(unboxed_closures)]
+#![feature(trace_macros)]
+#![feature(const_trait_impl)]
+#![feature(slice_as_chunks)]
+#![feature(variant_count)]
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+#![allow(clippy::bool_to_int_with_if)]
+#![allow(clippy::let_and_return)]
+// === Non-Standard Linter Configuration ===
+#![allow(clippy::option_map_unit_fn)]
+#![allow(clippy::precedence)]
+#![allow(dead_code)]
+#![deny(unconditional_recursion)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+use ensogl::prelude::*;
+use wasm_bindgen::prelude::*;
+
+use ensogl_core::application::Application;
+use ensogl_core::display::object::ObjectOps;
+use ensogl_hardcoded_theme as theme;
+
+
+
+// =====================
+// === ProjectsTable ===
+// =====================
+
+/// Module containing the [`ProjectsTable`], which is used to display a list of [`Project`]s in the
+/// Cloud dashboard.
+///
+/// The list of [`Project`]s, as rendered, contains information about the state of the [`Project`],
+/// the data associated with the [`Project`], the permissions assigned to it, etc. Buttons for
+/// interacting with the [`Project`]s (e.g., starting and stopping the [`Project`]s) are also
+/// rendered.
+///
+/// [`Project`]: ::enso_cloud_view::project::Project
+pub mod projects_table {
+    use ensogl::prelude::*;
+
+    use enso_cloud_view as view;
+    use ensogl::application;
+    use ensogl::data::color;
+    use ensogl_core::application::Application;
+    use ensogl_core::display;
+    use ensogl_core::frp;
+    use ensogl_grid_view::simple::EntryModel;
+    use ensogl_grid_view::Col;
+    use ensogl_grid_view::Row;
+
+
+
+    // =================
+    // === Constants ===
+    // =================
+
+    /// The width of a single entry in the [`ProjectsTable`], in pixels.
+    const ENTRY_WIDTH: f32 = 130.0;
+    /// The height of a single entry in the [`ProjectsTable`], in pixels.
+    const ENTRY_HEIGHT: f32 = 28.0;
+    /// A placeholder value for the initial height of the viewport. This value is a placeholder
+    /// because the value must be provided when creating the viewport, but is not used since the
+    /// viewport is resized to fit the [`ProjectsTable`] when the graphics context is initialized.
+    const VIEWPORT_HEIGHT: f32 = 300.0;
+    /// A placeholder value for the initial width of the viewport. This value is a placeholder
+    /// because the value must be provided when creating the viewport, but is not used since the
+    /// viewport is resized to fit the [`ProjectsTable`] when the graphics context is initialized.
+    const VIEWPORT_WIDTH: f32 = 400.0;
```

Comment:
ENTRY_WIDTH ENTRY_HEIGHT and GRID_HEIGHT_RATIO should be taken from the styles.
And isn't the VIEWPORT_WIDTH calculated basing on the scene shape anyway?